policy-engine/src/graph.rs

```diff
@@ -73,10 +71,18 @@ pub(crate) fn index(
         }))]
     };
 
-    let plugin_params = req.query().to_owned();
+    let plugin_params = req
+        .query_string()
+        .to_owned()
+        .split("&")
+        .map(|pair| {
+            let kv_split: Vec<&str> = pair.split("=").collect();
+            (kv_split[0].to_string(), kv_split[1].to_string())
```

Comment:
This will result in a client-controlled panic if `len < 2`, right?

Reply:
Indeed! where `len < 2` means the query argument doesn't contain a `=`. In that case we should probably give a meaningful error to the client.

Reply:
Since the processing of the query string isn't in the future chain it's not straight forward to generate an error for the user here. The current behavior in master is that such values are filled in with the empty string, so I''m opting for retaining that.

Reply:
@lucab let me know if you agree, then I'll squash the fixup commit.

Reply:
Ack. I agree it makes to either use an empty string, or filter-out valueless keys (assuming we'll never need a valueless key).